addons/mail/models/ir_attachment.py

```diff
@@ -71,5 +82,98 @@ def _attachment_format(self, commands=False):
                     'res_id': attachment.res_id,
                     'res_model': attachment.res_model,
                 })
-            res_list.append(res)
-        return res_list
+            attachments.append(res)
+        return attachments
+
+    @api.model
+    def _get_data_from_url(self, url):
+        """
+        This will create attachment data based on what we can read from the URL.
+        If the URL is an HTML page, this will parse the OpenGraph meta data.
+        If the URL is an image, this will create an image attachment.
+        """
+        try:
+            page = requests.get(url, timeout=1)
+        except requests.exceptions.RequestException:
+            return False
+
+        if page.status_code != requests.codes.ok:
+            return False
+
+        image_mimetype = [
+            'image/bmp',
+            'image/gif',
+            'image/jpeg',
+            'image/png',
+            'image/tiff',
+            'image/x-icon',
+        ]
+        if page.headers['Content-Type'] in image_mimetype:
+            return self._get_data_from_url_image(url)
+        elif 'text/html' in page.headers['Content-Type']:
+            return self._get_data_from_url_html(url, page.content)
+        return False
+
+    def _get_data_from_url_image(self, url):
+        image = self._get_image_from_url(url)
+        data = {
+            'url': url,
+            'name': url,
+            'description': False,
+            'raw': image,
+            'mimetype': 'image/o-linkpreview-image',
+        }
+        return data
+
+    def _get_image_from_url(self, image_url):
```

Comment:
Does getting an image from an URL = fetching an image? Maybe "fetch" would highlight a bit more that the method performs an HTTP request?
```suggestion
    def _fetch_image(self, image_url):
```